src/server/routes/scanner.js

```diff
@@ -1,13 +1,229 @@
+                };
+
+                categories.push(category);
+            }
+
+            category.rules.push(rule);
+
+            if (!excludedRuleStatus.includes(rule.status)) {
+                if (!category.results) {
+                    category.results = [];
+                }
+
+                category.results.push(rule);
+            }
+
+            if (includeIncompleteScan && rule.status === jobStatus.pending) {
+                if (!category.incompleteRules) {
+                    category.incompleteRules = [];
+                }
+
+                category.incompleteRules.push(rule);
+            }
+        });
+
+        categories = _.sortBy(categories, (category) => {
+            return category.name;
+        });
+
+        return categories;
+    };
+
+    /** Process scanning result to add category and statistics information */
+    const processRuleResults = (ruleResults, includeIncompleteScan = false) => {
+        const overallStatistics = {
+            errors: 0,
+            warnings: 0
+        };
+
+        const categories = parseCategories(ruleResults, includeIncompleteScan);
+
+        // Caculate numbers of `errors` and `warnings`.
+        _.forEach(categories, (category) => {
+            const statistics = _.reduce(category.results || [], (count, rule) => {
+                if (rule && rule.status === ruleStatus.error) {
+                    count.errors += rule.messages.length;
+                    overallStatistics.errors += rule.messages.length;
+                }
+
+                if (rule && rule.status === ruleStatus.warning) {
+                    count.warnings += rule.messages.length;
+                    overallStatistics.warnings += rule.messages.length;
+                }
+
+                return count;
+            }, { errors: 0, warnings: 0 });
+
+            category.statistics = statistics;
+        });
+
+        return { categories, overallStatistics };
+    };
+
     app.get('/scanner', (req, res) => {
         res.render('scan-form', {
             description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum',
-            title: 'Analyze your website using sonar online'
+            title: 'Scanner'
         });
     });
 
-    app.post('/scanner', (req, res) => {
-        res.render('scan-result', { title: 'Results for your page | sonar' });
+    app.get('/scanner/api/:id', async (req, res) => {
+        const id = req.params.id;
+        let scanResult;
+
+        try {
+            scanResult = await queryResult(id);
+        } catch (error) {
+            return res.status(404);
```

Comment:
Error shouldn't be `404` but rather a `500`. Also we should report the `error` somewhere.